edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/MetricsFileStorage.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class MetricsFileStorage : IMetricsStorage
+    {
+        readonly ISystemTime systemTime;
+
+        string directory;
```

Comment:
you may need to make it as public readonly for testing.

Reply:
this is just the value passed in the constructor

Reply:
mark it with readonly.